File: c#/Scripts/OGRuler.cs
using UnityEngine;
using System.Collections;

[ExecuteInEditMode]
[AddComponentMenu ("OpenGUI/Ruler")]
public class OGRuler : OGWidget {

	public Color color = Color.white;

	Texture2D bg;

	void Start (){
		bg = new Texture2D(2,2);
	}

	public override void Draw (  float x ,   float y   ){
		GUI.DrawTexture ( new Rect( x, y, transform.localScale.x, transform.localScale.y ), bg );
	}
	
	public override void UpdateWidget (){
		bg.SetPixels(new Color[] { color,color,color,color });
		bg.Apply();
	}
}

File: src/Scripts/OGRuler.js
#pragma strict

@script ExecuteInEditMode
@script AddComponentMenu ("OpenGUI/Ruler")

class OGRuler extends OGWidget {
	var color : Color = Color.white;

	@HideInInspector var bg : Texture2D;

	function Start () {
		bg = new Texture2D(2,2);
	}

	override function Draw ( x : float, y : float ) {
		GUI.DrawTexture ( Rect ( x, y, transform.localScale.x, transform.localScale.y ), bg );
	}
	
	override function UpdateWidget () {
		bg.SetPixels([color,color,color,color]);
		bg.Apply();
	}
}